### PR TITLE
Changed views

### PR DESCRIPTION
**Pull request by AmericanBurger**

### CODE REVIEW

File: mysite/settings.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 """
 
 import os
+SETTINGS_PATH = os.path.dirname(os.path.dirname(__file__))
```
Comment (agpl) [inline mysite/settings.py:14]: Is it still required, looks like redundant change

File: users/admin.py [hunk 2]
```diff
@@ -1,4 +1,8 @@
 from django.contrib import admin
-from .models import Thing
+from .models import *
 
 admin.site.register(Thing)
+
+def save_model(self, request, obj, form, change):
```
Comment (agpl) [inline users/admin.py:6]: Where it is used?

File: users/forms.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+from django import forms
+
+from .models import Thing
```
Comment (agpl) [inline users/forms.py:3]: It's better to use absolute imports

File: users/models.py [hunk 4]
```diff
@@ -1,8 +1,13 @@
...
+from django.contrib.auth.models import User
 
 class Thing(models.Model):
     name = models.CharField(max_length=60)
     bought_at = models.DateField('bought at')
-    
+    store = models.CharField(max_length=60)
+    created_at = models.DateTimeField(auto_now_add=True)
+    modified_at = models.DateTimeField(auto_now=True)
+    user = models.ForeignKey(User, editable=False, null=True, blank=True, on_delete=models.CASCADE)
```
Comment (agpl) [inline users/models.py:10]: I know, it's tricky to extend model with required foreign key. At this point you can delete local database and make it as required. Why it's marked as not editable?

File: users/views.py [hunk 5]
```diff
@@ -1,6 +1,61 @@
...
+            post.save()
+            return redirect('thing_detail', pk=post.pk)
+    else:
+        form = ThingPostForm(instance=post)
+    
+    return render(request, 'users/thing_edit.html', {'form': form})
+
+@login_required
+def thing_delete(request, pk):
+    query = Thing.objects.get(pk=pk)
```
Comment (agpl) [inline users/views.py:58]: Getting object in this file is not consistent. Please use the best method you used here

Comment (agpl) [reply users/views.py:58]: Choose better name for variable

File: users/views.py [hunk 6]
```diff
@@ -1,6 +1,61 @@
...
+    
+    return render(request, 'users/thing_detail.html', {'thing': thing})
+
+@login_required
+def thing_edit(request, pk):
+    post = get_object_or_404(Thing, pk=pk)
+    
+    if request.method == "POST":
+        form = ThingPostForm(request.POST, instance=post)
+        if form.is_valid():
```
Comment (agpl) [inline users/views.py:46]: Are you able to extract part of the code used here and before and make it as common method? DRY - do not repeat yourself

File: mysite/settings.py (hunk 1 above)
Comment (AmericanBurger) [reply mysite/settings.py:14]: It seems like it's not needed anymore, so I've deleted it.

File: users/forms.py (hunk 3 above)
Comment (AmericanBurger) [reply users/forms.py:3]: done :)

Comment (AmericanBurger) [reply users/forms.py:3]: done

File: users/admin.py (hunk 2 above)
Comment (AmericanBurger) [reply users/admin.py:6]: It's not compulsory either. Deleted it.

File: users/views.py (hunk 5 above)
Comment (AmericanBurger) [reply users/views.py:58]: already made it better

File: users/models.py (hunk 4 above)
Comment (AmericanBurger) [reply users/models.py:10]: fixed

File: users/views.py (hunk 6 above)
Comment (AmericanBurger) [reply users/views.py:46]: This is not the same. Those two uses are different.

**Pull request merged**